### ISSUE TITLE
Add strict type checking in tuple

### ISSUE THREAD

**Issue by namannimmo10** [lcompilers/lpython#433]
```python
def main():
  t: tuple[i32, str]
  t = (1, 2)

main()
```
This will generate an ASR, but actually should give an error that the second element is expected to be of `str` type.

**Comment by namannimmo10** [comment]
Same goes for dictionaries:
```python
def main():
  d: dict[i32, i32]
  d = {1: "a", 2: "b"}

main()
```

**Comment by yashsain** [comment]
hey, I'd like to work on this. I am relatively new to open source, so can you please guide me a bit as to how to proceed. Thanks

**Comment by certik** [comment]
See #439 which fixes something similar for lists. You have to implement a similar logic for tuples.
